app/docker/helpers/constraintsHelper.js

```diff
@@ -0,0 +1,107 @@
+function ConstraintModel(op, key, value) {
+  this.op = op;
+  this.value = value;
+  this.key = key;
+}
+
+var patterns = {
+  id: {
+    nodeId: 'node.id',
+    nodeHostname: 'node.hostname',
+    nodeRole: 'node.role',
+    nodeLabels: 'node.labels.',
+    engineLabels: 'engine.labels.'
+  },
+  op: {
+    eq: '==',
+    neq: '!='
+  }
+};
+
+function matchesConstraint(value, constraint) {
+  if (constraint && ((constraint.op === patterns.op.eq &&
+        value !== constraint.value) ||
+      (constraint.op === patterns.op.neq &&
+        value === constraint.value))) {
+    return false;
+  }
+  return true;
+}
+
+function matchesLabel(labels, constraint) {
+  if (!constraint) {
+    return true;
+  }
+  var found = labels.find(function (label) {
```

Comment:
The `find` function is not compatible with IE: https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/Array/find

Consider using `_.find()` instead: https://lodash.com/docs/4.17.10#find

As a general recommendation, please use lodash functions instead of generic JS methods as they usually less compatibility issues.